### PR TITLE
Onboarding Page

### PR DESCRIPTION
**Pull request by 0xbaggi**
I develop the onboarding page (#39), following the [figma](https://www.figma.com/file/6NyY9yqunpbU7HIkbNEAL3/Sossoldi-App?node-id=0%3A1) design:

done:
- "start the set up" page
- step 1 page (set budgets)
- Scrollable list of selectable budgets as a grid of cards
- "add category" card
- Dynamic style and layout (based on selected budget cards)
- step 2 page (set liquidity)
- Input fields for account name and ammount (with numeric keypad only.)

TODO: 
   - Show onboarding page only the first time

  (step 1 page)
  - Dynamic budgets from database (now they are static)
  - implement the function onTap on the "Add category" card
  
  (step 2 page)
  - implement the "edit icon and color" field input
  
https://user-images.githubusercontent.com/39710037/211912680-488d5880-1f44-4e19-b6e0-51dc7712b223.mp4

tested on Pixel XL API 31 Android 12

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -3,7 +3,7 @@ import 'dart:io';
 
 import 'package:flutter/material.dart';
 import 'routes.dart';
-import 'pages/structure.dart';
+import 'package:sossoldi/pages/onboarding_page.dart';
```
Comment (lucaantonelli) [inline lib/main.dart:6]: It's better to use relative path as mentioned in #43 . Check also the other files commited to update it.

Comment (FedericoBruzzone) [reply lib/main.dart:6]: Maybe the best choice is to use relative path :'D

**Review by FedericoBruzzone — CHANGES_REQUESTED** [review]

File: lib/pages/onboarding_page.dart [hunk 2]
```diff
@@ -0,0 +1,549 @@
...
+class _Step1State extends State<Step1> {
+  // static budget list for testing only
+  List<BudgetWrap> budgets = [
+    BudgetWrap("Home", 400, Colors.orange),
+    BudgetWrap("Free time", 100, Colors.blue),
+    BudgetWrap("Health", 100, Colors.pinkAccent),
+    BudgetWrap("Savings", 150, Colors.red),
+    BudgetWrap("Travel", 50, Colors.lightBlueAccent),
+    BudgetWrap("Shopping", 50, Colors.yellow),
+  ];
```
[the remark this answers is not in the document]
Comment (0xbaggi) [reply lib/pages/onboarding_page.dart:88]: Yes! I think the user will create his own custom budget and amount, but at the moment it is statically defined in this list (like names and colors).

File: lib/main.dart (hunk 1 above)
Comment (0xbaggi) [reply lib/main.dart]: Thank you! I just made a commit to fix this. 

I think there are others incorrectly named import in the project, maybe its needed a issue for check all files.

**Review by FedericoBruzzone — APPROVED** [review]

**Comment by FedericoBruzzone** [comment]
Can you open the issues regarding the things that are missing?

**Comment by 0xbaggi** [comment]
> Can you open the issues regarding the things that are missing?

Sure, #57
I think we should label it as a "good first issue"

**Comment by FedericoBruzzone** [comment]
Can I merge this staff? :D

**Comment by FedericoBruzzone** [comment]
Thanks for reminding me, write me when I can merge @mikev-cw

**Comment by 0xbaggi** [comment]
@FedericoBruzzone @mikev-cw 
Can we merge this now?

**Comment by FedericoBruzzone** [comment]
> @FedericoBruzzone @mikev-cw 
> 
> Can we merge this now? 

I'm waiting for them to tell me to proceed

**Comment by mikev-cw** [comment]
@0xbaggi merging of #63 has raised some expected conflicts. Can you check if you can easily resolve them? Head on Discord if you need help

**Comment by FedericoBruzzone** [comment]
You should update your fork, after that you can use this command locally:
```
git switch main
git pull main
git switch -
git rebase main
``` 
and at this point resolve conflicts in your favorite text editor.

Please, don't create another commit, after the above command use `git commit --amend & git push --force` to update this pull request.

**Pull request closed without merging**